### PR TITLE
added first party to java enforcer

### PR DESCRIPTION
**Pull request by nitzanpx**

### CODE REVIEW

File: src/main/java/com/perimeterx/api/PerimeterX.java [hunk 1]
```diff
@@ -197,6 +207,16 @@ public PXContext pxVerify(HttpServletRequest req, HttpServletResponseWrapper res
         return context;
     }
 
+    private boolean shouldReverseRequest(HttpServletRequest req, HttpServletResponseWrapper res) throws Exception {
+        if (reverseProxy.reversePxClient(req, res)){
+            return true;
+        } else if (reverseProxy.reversePxXhr(req, res)) {
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/PerimeterX.java]: the `else` is not redundant

File: src/main/java/com/perimeterx/api/proxy/DefaultPredefinedResponseHandler.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+
+public class DefaultPredefinedResponseHandler implements PredefinedResponseHelper {
+    PXLogger logger = PXLogger.getLogger(DefaultPredefinedResponseHandler.class);
+
+    @Override
+    public void handlePredefinedResponse(HttpServletResponse res, PredefinedResponse predefinedResponse) {
+        try {
+            res.setHeader(HttpHeaders.CONTENT_TYPE, predefinedResponse.getContentType());
+            res.setHeader("X-PX-RESPONSE", "PREDEFINED");
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/proxy/DefaultPredefinedResponseHandler.java]: remove

File: src/main/java/com/perimeterx/api/proxy/DefaultReverseProxy.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+        cm.setMaxTotal(pxConfiguration.getMaxConnections());
+        cm.setDefaultMaxPerRoute(pxConfiguration.getMaxConnectionsPerRoute());
+        this.proxyClient = HttpClients.custom()
+                .setConnectionManager(cm)
+                .build();
+    }
+
+
+    public boolean reversePxClient(HttpServletRequest req, HttpServletResponseWrapper res) throws Exception {
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/proxy/DefaultReverseProxy.java]: please add documentation

File: src/main/java/com/perimeterx/api/proxy/RemoteServer.java [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+
+    public HttpResponse handleResponse(HttpRequest proxyRequest, boolean allowPredefinedHandler) {
+        HttpResponse proxyResponse = null;
+        try {
+            // Execute the request
+            proxyResponse = doExecute(proxyRequest);
+
+            // Process the response:
+
+            // Pass the response code. This method with the "reason phrase" is deprecated but it's the
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/api/proxy/RemoteServer.java]: comment is no longer relevant

File: src/main/java/com/perimeterx/api/PerimeterX.java (hunk 1 above)
Comment (nitzanpx) [reply src/main/java/com/perimeterx/api/PerimeterX.java]: 👍

File: src/main/java/com/perimeterx/api/proxy/DefaultPredefinedResponseHandler.java (hunk 2 above)
Comment (nitzanpx) [reply src/main/java/com/perimeterx/api/proxy/DefaultPredefinedResponseHandler.java]: 👍

File: src/main/java/com/perimeterx/api/proxy/DefaultReverseProxy.java (hunk 3 above)
Comment (nitzanpx) [reply src/main/java/com/perimeterx/api/proxy/DefaultReverseProxy.java]: 👍 added docs to interface

File: src/main/java/com/perimeterx/api/proxy/RemoteServer.java (hunk 4 above)
Comment (nitzanpx) [reply src/main/java/com/perimeterx/api/proxy/RemoteServer.java]: 👍

File: src/main/java/com/perimeterx/api/proxy/RemoteServer.java [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+            servletCookie.setVersion(cookie.getVersion());
+            servletResponse.addCookie(servletCookie);
+        }
+    }
+
+    /**
+     * Copy request headers from the servlet client to the proxy request.
+     * This is easily overridden to add your own.
+     */
+    protected void copyRequestHeaders(HttpServletRequest servletRequest, HttpRequest proxyRequest) {
```
[the remark this answers is not in the document]
Comment (nitzanpx) [reply src/main/java/com/perimeterx/api/proxy/RemoteServer.java]: 👍

**Pull request merged**